src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Cache.cs

```diff
@@ -228,88 +275,60 @@ private static bool TryGetCacheValueSmall(CachedCodeEntryKey key, [NotNullWhen(t
             return entry;
         }
 
-        /// <summary>
-        /// Used as a key for CacheCodeEntry
-        /// </summary>
-        internal readonly struct CachedCodeEntryKey : IEquatable<CachedCodeEntryKey>
+        /// <summary>Used as a key for <see cref="Node"/>.</summary>
+        internal readonly struct Key : IEquatable<Key>
         {
             private readonly string _pattern;
-            private readonly string _cultureKey;
+            private readonly string _culture;
             private readonly RegexOptions _options;
             private readonly bool _hasTimeout;
 
-            public CachedCodeEntryKey(string pattern, string cultureKey, RegexOptions options, bool hasTimeout)
+            public Key(string pattern, string culture, RegexOptions options, bool hasTimeout)
             {
-                SysDebug.Assert(pattern != null, "Pattern must be provided");
-                SysDebug.Assert(cultureKey != null, "Culture must be provided");
+                Debug.Assert(pattern != null, "Pattern must be provided");
+                Debug.Assert(culture != null, "Culture must be provided");
 
                 _pattern = pattern;
-                _cultureKey = cultureKey;
+                _culture = culture;
                 _options = options;
                 _hasTimeout = hasTimeout;
             }
 
             public override bool Equals(object? obj) =>
-                obj is CachedCodeEntryKey other && Equals(other);
+                obj is Key other && Equals(other);
 
-            public bool Equals(CachedCodeEntryKey other) =>
+            public bool Equals(Key other) =>
                 _pattern.Equals(other._pattern) &&
-                _cultureKey.Equals(other._cultureKey) &&
+                _culture.Equals(other._culture) &&
                 _options == other._options &&
                 _hasTimeout == other._hasTimeout;
 
-            public static bool operator ==(CachedCodeEntryKey left, CachedCodeEntryKey right) =>
+            public static bool operator ==(Key left, Key right) =>
                 left.Equals(right);
 
-            public static bool operator !=(CachedCodeEntryKey left, CachedCodeEntryKey right) =>
+            public static bool operator !=(Key left, Key right) =>
                 !left.Equals(right);
 
             public override int GetHashCode() =>
                 _pattern.GetHashCode() ^
```

Comment:
`HashCode.Combine` ? 

Although in this case, perf is far more important than entropy. Plus, the problem with `^` that it coalesces nulls isn't an issue these won't be null.